src/python/pants/backend/javascript/nodejs_project.py

```diff
@@ -236,6 +236,8 @@ async def find_node_js_projects(
     nodejs: NodeJS,
     resolve_names: UserChosenNodeJSResolveAliases,
 ) -> AllNodeJSProjects:
+    # Note: If pnpm_workspace.yaml is present for an npm-managed project, it will override the package.json["workspaces"] setting, which is not intuitive
```

Comment:
I took the liberty of creating https://github.com/pantsbuild/pants/issues/21134, please link it in this comment.

Reply:
Added it now.